### DISCUSSION TITLE
How can I convert C# arrays to JS arrays?

### DISCUSSION THREAD

**Discussion by Soundman32** [ClearFoundry/ClearScript#601] [Q&A] [answered]
I have a C# method that returns a collection:

```csharp
public IList GetItems()
{
   ...
   return collection.ToArray();
}
```

In the JS code, I can't use `.length` to get the length of the array, but I can use `.Count` (which is obviously the C# object property).

I also want to iterate over this collection, but this also fails.

```js
    const items = _myObject.GetItems();
    items.forEach(async item => { } );
```

**_How can I convert C# arrays to JS arrays?_**

**Comment by ClearScriptLib** [comment, marked as answer]
Hello @Soundman32,

>I also want to iterate over this collection, but this also fails.

Enumerable .NET objects are [iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) from JavaScript, so you can do this:

```JavaScript
for (const item of _myObject.GetItems()) {
    // do something
}
```

>How can I convert C# arrays to JS arrays?

Because .NET arrays are iterable, you can use JavaScript's [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from). You can even call it directly from the host:

```csharp
var jsArray = engine.Script.Array.from(dotNetArray);
```

Good luck!
